### DISCUSSION TITLE
Installation problem 2FA setup & Bad Gateway error - Ubuntu 20.04

### DISCUSSION THREAD

**Discussion by 10YAR** [deeztek/Hermes-Secure-Email-Gateway#61] [Support] [answered]
Hello,

I tried to install Hermes SEG on my fresh Ubuntu 22 install. It said that the installation finished and I've access to the server and can login. Problem is at the 2FA setup step, there's an error showing. See the screenshot below :

![CleanShot 2024-01-05 at 17 29 28@2x](https://github.com/deeztek/Hermes-Secure-Email-Gateway/assets/76828073/6cfa5812-ee9f-42ec-9b32-c65a6faf04c8)

I also add the installation logs so that you see if there's a problem.
[install_log-01-05-2024-1539.log](https://github.com/deeztek/Hermes-Secure-Email-Gateway/files/13844266/install_log-01-05-2024-1539.log)

Thank you

Edit: Tried on 20.04 after, same problem

**Comment by deeztek** [comment]
Hermes will not install correctly on 22.04 yet. Please see Installation notes for the requirements:

[https://github.com/deeztek/Hermes-Secure-Email-Gateway?tab=readme-ov-file#installation](https://github.com/deeztek/Hermes-Secure-Email-Gateway?tab=readme-ov-file#installation)

We will be fixing this in the near future, but for now, 20.04 is recommended.

Thanks

**Comment by 10YAR** [comment]
@deeztek Thank you,
so I reinstalled a fresh Ubuntu 20.04 and did the installation but same problem...
Here's the log file

[install_log-01-05-2024-1912.log](https://github.com/deeztek/Hermes-Secure-Email-Gateway/files/13845844/install_log-01-05-2024-1912.log)

Apparently, there was two fails during installation :
![CleanShot 2024-01-05 at 20 35 43@2x](https://github.com/deeztek/Hermes-Secure-Email-Gateway/assets/76828073/cdf821ef-87ea-4c78-b34c-eec8f494e9fd)


Also I have to mention that after the admin login there's a " Bad Gateway " error.

Also I have installed mariadb just before launching the installation of Hermes SEG. I wonder if it's the correct way to do it or if I shouldn't have installed it (maybe the installation script of Hermes already installs mariadb? I don't know).

**Comment by deeztek** [comment]
Sorry for the delay. The installation errors you are getting usually are related to attempting to install on a LXC environment. Is this a vm or bare metal hardware?

**Comment by deeztek** [comment]
> Also I have installed mariadb just before launching the installation of Hermes SEG. I wonder if it's the correct way to do it or if I shouldn't have installed it (maybe the installation script of Hermes already installs mariadb? I don't know).

Oops just noticed this comment. Yes, mariadb should not be installed beforehand. The install script does all that for you.

**Comment by 10YAR** [comment]
> Sorry for the delay. The installation errors you are getting usually are related to attempting to install on a LXC environment. Is this a vm or bare metal hardware?

It's a bare metal hardware



> Oops just noticed this comment. Yes, mariadb should not be installed beforehand. The install script does all that for you.

Should I retry on a fresh ubuntu 20.04 install ?


Thank you 🙏🏼

**Comment by deeztek** [comment]
That would be the safest option.  Thanks

**Comment by 10YAR** [comment]
> That would be the safest option. Thanks

So I reinstalled it, but the same problem is still present: Bad gateway error on /admin route and 2FA setup error on button click on index route... I'm adding log files

Also I was wondering, would 2FA work better if I setup a SSL cert in the nginx configuration with letsencrypt ?

[install_log-01-06-2024-2101.log](https://github.com/deeztek/Hermes-Secure-Email-Gateway/files/13851455/install_log-01-06-2024-2101.log)

**Comment by deeztek** [comment]
Did you get any  failures on the install screen?

**Comment by deeztek** [comment, marked as answer]
> So I reinstalled it, but the same problem is still present: Bad gateway error on /admin route and 2FA setup error on button click on index route... I'm adding log files
> 

Sorry, not sure why I didn't think of this earlier. We had this issue two weeks ago and it ended up being the commandbox server.json file. Ortus changed their versioning scheme from 5.8 to 5.9 so that was breaking the commandbox installation. I have gone ahead and fixed the file in the repo with the correct versioning scheme. Please re-clone the repo and start fresh and it should work.

Thanks

**Comment by 10YAR** [comment]
It works ! Thank you very much
![CleanShot 2024-01-08 at 00 09 38@2x](https://github.com/deeztek/Hermes-Secure-Email-Gateway/assets/76828073/c6172a99-dd72-4a91-8ce6-8d068068c626)

**Comment by deeztek** [comment]
Glad you got it working!
